scipy/interpolate/tests/test_bary_rational.py

```diff
@@ -81,7 +81,7 @@ def test_integer_promotion(method, dtype):
 
 class TestAAA:
     def test_input_validation(self):
-        with pytest.raises(ValueError, match="same size"):
+        with pytest.raises(ValueError, match="size"):
```

Comment:
A little more specific message check would be good, if only to make it obvious where this is coming from. While we're at it, maybe manually use the whole error message, including numbers? I've seen a few cases where this would find that an f-string wasn't actually evaluated because it wasn't actually marked to be an f-string.